pandas/_libs/index.pyx

```diff
@@ -599,70 +600,72 @@ cpdef convert_scalar(ndarray arr, object value):
     return value
 
 
-cdef class MultiIndexObjectEngine(ObjectEngine):
-    """
-    provide the same interface as the MultiIndexEngine
-    but use the IndexEngine for computation
+cdef class BaseMultiIndexCodesEngine(object):
+    def __init__(self, levels, labels, offsets, **kwargs):
+        self._levels = levels
+        self._offsets = offsets
 
-    This provides good performance with samller MI's
-    """
-    def get_indexer(self, values):
-        # convert a MI to an ndarray
-        if hasattr(values, 'values'):
-            values = values.values
-        return super(MultiIndexObjectEngine, self).get_indexer(values)
+        # Map each combination to an integer
+        codes = (np.array(labels, dtype='int64').T + 1).astype('uint64')
+        lab_ints = self._codes_to_ints(codes)
```

Comment:
de-privatize method names, spell them out, e.g.``codes_to_integer``

Reply:
This is is the only method name starting with ``_`` as it's not meant to be used from outside (like e.g. ``IndexEngine._maybe_get_bool_indexer``). Why should it not be private?

Reply:
I understand but that a historic effect. These classes themselves are totally private, so making the methods is overkill.

Reply:
I thought it would make it clear that e.g. ``BaseMultiIndexCodesEngine.get_loc`` is meant to be used from ``Index`` (internal) code, while ``BaseMultiIndexCodesEngine._codes_to_ints`` is only meant to be used from code specific to the engine. But fine, will change.

Reply:
I agree with @toobaz: if something is a helper method that is only used within the class, make this clear by using a `_...` name (also for within the codebase is it useful to know which method is 'internally' public and which not, even if the class itself is fully private)